doc/xml/firewalld.richlanguage.xml

```diff
@@ -280,12 +280,35 @@ log [prefix="prefix text"] [level="log level"] [limit value="rate/duration"]
       </para>
       <para>
 	Log new connection attempts to the rule with kernel logging for example in syslog.
-	You can define a prefix text that will be added to the log message as a prefix.
+	You can define a prefix text with a maximum length of 29 characters that will be added to the log message as a prefix.
 	Log level can be one of "<option>emerg</option>", "<option>alert</option>", "<option>crit</option>", "<option>error</option>", "<option>warning</option>", "<option>notice</option>", "<option>info</option>" or "<option>debug</option>", where default (i.e. if there's no one specified) is "<option>warning</option>". See <citerefentry><refentrytitle>syslog</refentrytitle><manvolnum>3</manvolnum></citerefentry> for description of levels.
 	See Limit section for description of <option>limit</option> tag.
       </para>
     </refsect2>
 
+    <refsect2>
+      <title>NFLog</title>
+      <para>
+	<programlisting>
+nflog [group="group id"] [prefix="prefix text"] [queue-size="threshold"] [limit value="rate/duration"]
+	</programlisting>
+      </para>
+      <para>
+	Log new connection attempts to the rule using kernel logging to pass the packets through a 'netlink' socket to users or applications monitoring the multicast <option>group</option>.
+      The minimum and default value for <option>group</option> is 0, maximum value is 65535.
+	See NETLINK_NETFILTER in <citerefentry><refentrytitle>netlink</refentrytitle><manvolnum>7</manvolnum></citerefentry> man page and NFLOG in both <citerefentry><refentrytitle>iptables-extensions</refentrytitle><manvolnum>8</manvolnum></citerefentry> and <citerefentry><refentrytitle>nft</refentrytitle><manvolnum>8</manvolnum></citerefentry> man pages for a more detailed description.
+      </para>
+      <para>
+      You can define a prefix text with a maximum length of 29 characters that will be added to the log message as a prefix.
```

Comment:
So after going over the source code, man pages and testing, I've discovered some discrepancies and I'll address each target and limitations starting with `NFLOG`.

In iptables-extensions(8) man pages under target NFLOG it mentions that the prefix can be at most 64 characters in length. That is incorrect, the prefix is truncated after 63 characters and that can be verified in the source [here](https://github.com/torvalds/linux/blob/34816d20f173a90389c8a7e641166d8ea9dce70a/include/uapi/linux/netfilter/xt_NFLOG.h#L22) and through testing.

Reply:
Does nftables have the same restriction? What you linked is `XT`, i.e. xtables (iptables).

Reply:
I apologize, nftables has a 127 character limit which is defined [here](https://github.com/torvalds/linux/blob/fcadab740480e0e0e9fa9bd272acd409884d431a/include/uapi/linux/netfilter/nf_log.h#L13). Anything over the limit gives a `Error: Could not process rule: Numerical result out of range`.

Reply:
Okay. I think 63 is enough. The iptables will be deprecated in the next release. We can increase the limit in the future if needed. Maybe note this via a comment near where you're clamping the length.

Reply:
Sounds good, I'll add that soon.

Reply:
Another options is to clamp both log and nflog at 127 and just document the silent truncation for iptables backend. Since there is no issue storing or matching a longer prefix since no error is thrown from iptables.

Reply:
Find by me. The iptables backend will be marked deprecated in the next major release anyways. I don't want to artificially limit nftables due to iptables.

Reply:
I agree.